src/App.tsx

```diff
@@ -15,48 +15,96 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  NAME = 'name',
+  LENGTH = 'length',
+  NONE = 'none',
+}
+
+function sortArray(
+  goods: string[],
+  howSort: SortType,
+  isReversed: boolean,
+) {
+  const array = [...goods];
+
+  array.sort((value1, value2) => {
+    switch (howSort) {
+      case SortType.NAME:
+        return value1.localeCompare(value2);
+
+      case SortType.LENGTH:
+        return value1.length - value2.length;
+
+      default:
+        return 0;
+    }
+  });
+
+  if (isReversed) {
+    array.reverse();
+  }
+
+  return array;
+}
+
 export const App: React.FC = () => {
+  const [sort, setSort] = useState(SortType.NONE);
+  const [isReversed, setIsReversed] = useState(false);
+  const visibleGoods = sortArray(
+    goodsFromServer,
+    sort,
+    isReversed,
+  );
+
   return (
     <div className="section content">
       <div className="buttons">
         <button
           type="button"
-          className="button is-info is-light"
+          className={`button is-info ${sort === SortType.NAME ? null : 'is-light'}`}
+          onClick={() => setSort(SortType.NAME)}
         >
           Sort alphabetically
         </button>
 
         <button
           type="button"
-          className="button is-success is-light"
+          className={`button is-success ${sort === SortType.LENGTH ? null : 'is-light'}`}
+          onClick={() => setSort(SortType.LENGTH)}
         >
           Sort by length
         </button>
 
         <button
           type="button"
-          className="button is-warning is-light"
+          className={`button is-warning ${isReversed === true ? null : 'is-light'}`}
```

Comment:
Consider will be better if you use classnames library
```suggestion
          className={`button is-warning ${isReversed ? null : 'is-light'}`}
```